packages/core/admin/ee/server/services/review-workflows/workflows/index.js

```diff
@@ -49,30 +50,32 @@ module.exports = ({ strapi }) => {
      * @throws {ValidationError} - If the workflow has no stages.
      */
     async create(opts) {
-      let createOpts = { ...opts, populate: { stages: true } };
-
-      if (!opts.data.stages || opts.data.stages.length === 0) {
-        throw new ValidationError('Can not create a workflow without stages');
-      }
+      workflowsValidationService.validateWorkflowStages(opts.data.stages);
+      await workflowsValidationService.validateWorkflowCount(1);
 
       return strapi.db.transaction(async () => {
-        // Create stages
-        const stageIds = await getService('stages', { strapi })
-          .replaceStages([], opts.data.stages)
-          .then((stages) => stages.map((stage) => stage.id));
+        // Create Workflow
+        const workflow = await strapi.entityService.create(WORKFLOW_MODEL_UID, {
+          select: '*',
+          data: omit('stages')(opts.data),
+        });
 
-        createOpts = set('data.stages', stageIds, createOpts);
+        const stagesToCreate = opts.data.stages.map((stage) => ({
+          ...stage,
+          workflow: workflow.id,
+        }));
+        // Create stages
+        const stages = await getService('stages', { strapi }).createMany(stagesToCreate);
 
         // Update (un)assigned Content Types
         if (opts.data.contentTypes) {
           await workflowsContentTypes.migrate({
             destContentTypes: opts.data.contentTypes,
-            stageId: stageIds[0],
+            stageId: stages[0].id,
           });
         }
 
-        // Create Workflow
-        return strapi.entityService.create(WORKFLOW_MODEL_UID, createOpts);
+        return { ...workflow, stages };
```

Comment:
I don't see why this change is necessary if you are already validating the worklfow number and number of stages before anything else.